packages/trezor-user-env-link/src/api.ts

```diff
@@ -22,10 +22,33 @@ export interface StartEmu {
     model?: Model;
 }
 
-interface StartEmuFromUrl extends Omit<StartEmu, 'version'> {
+interface StartEmuType extends StartEmu {
+    type?: 'emulator-start';
```

Comment:
should `type` be optional here?

Reply:
also for `StartEmuType` should wipe and model be optional?

Reply:
`type` needs to be optional here so everything is working in runtime and is `typescript`-correct in `packages/connect/e2e/common.setup.ts`. Crucial lines are below:
```ts
const emulatorStartOpts: EmuStartOptsType =
    (process.env.emulatorStartOpts as any) || global.emulatorStartOpts || {};
const emuStartType = emulatorStartOpts.type;

switch (emuStartType) {
  case 'emulator-start':
  case undefined:
   ...
```

For runtime, it is possible that the `process.env.emulatorStartOpts` is empty, so it will try to take `global.emulatorStartOpts` ... I did not know where it is defined, I see only the interface:
```ts
    var emulatorStartOpts: {
        version: string;
        model: 'T1B1' | 'T2T1' | 'T2B1' | 'T3T1';
    };
```

So one possibility is to add `type: 'emulator-start'` there, and the second one was to make it optional and account for the `undefined` in switch.

Reply:
Good point, in `StartEmuType` almost all parameters are needed - `version`, `model`, `wipe` - that probably means not extending from `StartEmu`, as it overrides almost everything

Reply:
Correction - it was now possible to get rid of the `type` optionality - [3eddfc5](https://github.com/trezor/trezor-suite/pull/15689/commits/3eddfc5552f23084494f7a8c624e4a47509b3965)

However, now the `version` needs to be optional, because it is potentially `undefined` in this case:
```ts
const latest = getLatestFirmware(model);
const version = firmwareArg
    ? firmwareArg?.endsWith('-latest')
        ? latest
        : firmwareArg
    : latest;
emulatorStartOpts = {
    type: 'emulator-start',
    wipe: true,
    version,
    model,
};
```